### PR TITLE
Fix app lifecycle

### PR DESCRIPTION
**Pull request by Hornwitser**
Moves WebSocket server into fastify server and fixes the open handles left open in the test environment.

### CODE REVIEW

File: test/prepareTest.ts [hunk 1]
```diff
@@ -2,14 +2,20 @@ import ENV from "../src/utils/env.js"
...
+		ignoreUndefined: true,
+		loggerLevel: "info"
+	})
 
-export default backend
+	await backend.listen(0);
+})
+
+afterAll(async () => {
+	await backend.close()
```
Comment (oof2win2) [inline test/prepareTest.ts:16]: Same here
```suggestion
```

File: test/prepareTest.ts [hunk 2]
```diff
@@ -2,14 +2,20 @@ import ENV from "../src/utils/env.js"
...
-}) // connect to db before loading other stuff
+jest.setTimeout(10000)
+beforeAll(async () => {
+	await mongoose.connect(ENV.MONGOURI, {
+		ignoreUndefined: true,
+		loggerLevel: "info"
+	})
 
-export default backend
+	await backend.listen(0);
```
Comment (oof2win2) [inline test/prepareTest.ts:12]: This doesn't need to be there, because `fastify.inject` works even if the backend is not listening
```suggestion
```

Comment (Hornwitser) [reply test/prepareTest.ts:12]: Does the WebSocket server work without listening on a port? Does fastify.inject handle upgrade requests in a way that new WebSocket(address) can be redirected to? If not then the backend will have to listen on a port.

Comment (oof2win2) [reply test/prepareTest.ts:12]: Hmm, not sure about the WebSocket server. The thing is, fastify.inject is supposed to simulate requests only - run handlers without actually listening on any port. I guess it won't work, so you are probably right. Will need to test it when the WS tests are added.

**Pull request merged**